linera-execution/src/execution_state_actor.rs

```diff
@@ -360,6 +362,400 @@ where
 
         Ok(())
     }
+
+    pub(crate) async fn handle_requests(
+        &mut self,
+        execution_state_receiver: &mut UnboundedReceiver<ExecutionRequest>,
+    ) -> Result<(), ExecutionError> {
+        use ExecutionRequest::*;
+
+        let request_handle = |request, this: &'static RwLock<&'static mut Self>| async {
+            match request {
+                #[cfg(not(web))]
+                LoadContract { id, callback } => {
+                    let actor = this.read().await;
+                    let  response = actor.load_contract(id).await?;
+                    drop(actor);
```

Comment:
There's a pattern here that we can simplify and remove the `drop` by doing:

```suggestion
                    let response = this.read().await.load_contract(id).await?;
```